File: JS/EntradaSaidaLendoPulandoNomes.test.js
/**
 * @version JAVASCRIPT (ES6)
 * @author MARIO SAM <eu@mariosam.com.br>
 * @see I would love to work with you instead solving web code tests: hire me!
 * $ node NomeDaClasse.test
 */
const TEST = require('tape')
const OBJ = require('./EntradaSaidaLendoPulandoNomes.js')

TEST('Starting EntradaSaidaLendoPulandoNomes test...', (t) => {
    //Test 1
    let want = ["UFCG", "ITA", "URI"]
    let got = OBJ.giveMe379( ["USP","UFPE","UFCG","UFRN","UFRJ","IME","ITA","UNIOESTE","URI","UFG"] )
    t.assert(want.toString === got.toString, "Expect: UFCG ITA URI")
    //Test 2
    want = ["UNIFEI","UFRGS","UFU"]
    got = OBJ.giveMe379( ["UnB","UFMG","UNIFEI","UECE","UNICAMP","INATEL","UFRGS","UNIFESO","UFU","PUC"] )
    t.assert(want.toString === got.toString, "Expect: UNIFEI UFRGS UFU")

    t.end()
})

File: JAVA/EntradaSaidaLendoPulandoNomesTest.java
/** 
 * @version JAVA 1.8
 * @author MARIO SAM <eu@mariosam.com.br>
 * @see I would love to work with you instead solving web code tests: hire me!
 * $ mvn clean test -Dtest=your.package.TestClassName
 */ 
package JAVA;

import static org.junit.Assert.assertArrayEquals;
import org.junit.Test;

public class EntradaSaidaLendoPulandoNomesTest {

	@Test
	public void testEntradaSaidaLendoPulandoNomes() throws Exception {
		//Test 1
		String[] want = {"UFCG", "ITA", "URI"};
		String[] got = EntradaSaidaLendoPulandoNomes.giveMe379( new String[]{"USP","UFPE","UFCG","UFRN","UFRJ","IME","ITA","UNIOESTE","URI","UFG"} );
        assertArrayEquals(want, got);
		//Test 2
		want = new String[] {"UNIFEI","UFRGS","UFU"};
		got = EntradaSaidaLendoPulandoNomes.giveMe379( new String[]{"UnB","UFMG","UNIFEI","UECE","UNICAMP","INATEL","UFRGS","UNIFESO","UFU","PUC"} );
        assertArrayEquals(want, got);
	}

}
